gateway/ln-gateway/src/lib.rs

```diff
@@ -1277,7 +1270,13 @@ impl Gateway {
             }
 
             for (federation_id, federation_config) in federations {
-                if let Some(client) = self.clients.read().await.get(federation_id) {
+                if let Some(client) = self
+                    .federation_manager
+                    .clients
```

Comment:
Love the idea of a `FederationManager` and a `LightningManager`. Can we abstract away some of these operations so that their internal fields are not exposed? Something like `self.federation_manager.get_client(federation_id)`? (not sure if you did this in a separate commit or are planning on a separate PR

Reply:
💯 agree, the data in these structs should ideally be silo'd from the rest of the gateway

Planning this in a follow-up PR!